ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -692,3 +722,140 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
+	logger.Info(fmt.Sprintf("%s command result: %s", tcCheckInjectionCommandComposed, string(cmdOutput[:])))
+	var outputUnmarshalled []map[string]interface{}
+	err = json.Unmarshal(cmdOutput, &outputUnmarshalled)
+	if err != nil {
+		return false, errors.New("failed to unmarshal tc command output: " + err.Error())
+	}
+	netemExists := false
+	for _, line := range outputUnmarshalled {
+		// First check field "kind":"netem" exists.
+		if line["kind"] == "netem" {
+			// Now check if field "loss":"<loss percent>" exists, and if the percentage matches with the value in the request.
+			if options := line["options"]; options != nil {
+				if lossRandom := options.(map[string]interface{})["loss-random"]; lossRandom != nil {
+					if loss := lossRandom.(map[string]interface{})["loss"]; loss != nil {
+						if lossValue, ok := loss.(float64); ok {
+							lossPercentInPercentage := float64(*lossPercent) / 100
+							if lossValue == lossPercentInPercentage {
+								netemExists = true
+								break
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	// If we didn't find anything from above, there's no fault injected.
+	if !netemExists {
+		return false, nil
+	}
+
+	// Now check if the desired IPs are properly added to the filters.
+	// We run the following command: "tc -j filter show dev <INTERFACE>"
+	// The output of the command above does not format into json properly.
+	// It has a field like this: "options":{something here}
+	// Due to the field above, we won't be able to properly unmarshal this.
+	// Thus, we will use strings.contains directly to parse the output.
+	tcCheckIPCommandComposed := nsenterPrefix + fmt.Sprintf(tcCheckIPFilterCommandString, interfaceName)
+	cmdOutput, err = h.runExecCommand(ctx, tcCheckIPCommandComposed)
+	if err != nil {
+		return false, errors.New("failed to check network-packet-loss-fault: " + string(cmdOutput[:]) + err.Error())
+	}
+	// Log the command output to better help us debug.
+	logger.Info(fmt.Sprintf("%s command result: %s", tcCheckIPCommandComposed, string(cmdOutput[:])))
+	allIPAddressesInRequestExist := true
+	for _, ipAddress := range ipSources {
+		ipAddressInHex, err := convertIPAddressToHex(*ipAddress)
+		if err != nil {
+			return false, errors.New("failed to check network-packet-loss-fault: " + err.Error())
+		}
+		patternString := "match " + ipAddressInHex
+		if !strings.Contains(string(cmdOutput[:]), patternString) {
+			allIPAddressesInRequestExist = false
+			break
+		}
+	}
+	if !allIPAddressesInRequestExist {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// runExecCommand wraps around the execwrapper, providing a convenient way of running any Linux command
+// and getting the result in both stdout and stderr.
+func (h *FaultHandler) runExecCommand(ctx context.Context, linuxCommandString string) ([]byte, error) {
+	commandArray := strings.Split(linuxCommandString, " ")
+	cmdExec := h.osExecWrapper.CommandContext(ctx, commandArray[0], commandArray[1:]...)
```

Comment:
Q: Should there be a `cmdExec.Run()` call somewhere in this method?

Reply:
`cmdExec.CombinedOutput()` has output signature `([]byte, error)`, which is the same as the output of this method. We return the result of `cmdExec.CombinedOutput()` directly. CombinedOutput() is a generic method is os/exec. It runs the command and gives the stdout and stderr result at the same time (whereas Output() will only give stdout when exit code was 0, otherwise stdout will be nil)

Reply:
Ah I see now